### PR TITLE
Update to Consortium/main

### LINKED ISSUE #71: update to latest Consortium/main
Updates to Consortium/main include the following

- https://github.com/CICE-Consortium/CICE/pull/849. This PR implements improvements to the areafact calculation for remapping. This PR will change baselines (as expected). As noted in the 6.5 release notes, this improves C-grid solutions

- https://github.com/CICE-Consortium/CICE/pull/902. This PR includes an update to Icepack to use ``-puny`` when limiting fluxes (https://github.com/CICE-Consortium/Icepack/pull/474). This PR will change baselines (as expected).

- https://github.com/CICE-Consortium/CICE/pull/889. This PR adds a new name list configuration ``cpl_frazil`` to address an issue first noted in https://github.com/CICE-Consortium/Icepack/issues/390. In UWM the default configuration when coupled to FV3atm is ``ktherm=2`` and ``update_ocn_f=true``, meaning the salt and freshwater fluxes are computed in CICE and passed to MOM6. For this configuration, the default of ``cpl_frazil`` will be ``fresh_ice_corr``. Testing has shown this PR alone does not change answers in UWM for either the ``ktherm=1`` or ``ktherm=2`` cases.

### PR DESCRIPTION
**Pull request by DeniseWorthen**
- fixes #71 

Testing for UWM shows any test using CICE will change results see https://github.com/ufs-community/ufs-weather-model/pull/2027

### CODE REVIEW

File: cicecore/cicedyn/analysis/ice_history.F90 [hunk 1]
```diff
@@ -3656,501 +3654,29 @@ subroutine accum_hist (dt)
...
-                             a2D(i,j,n_siforceintstry(ns),iblk)*avgct(ns)*ravgip(i,j)
-                       endif
-                       if (ravgip(i,j) == c0) a2D(i,j,n_siforceintstry(ns),iblk) = spval_dbl
-                    enddo             ! i
-                    enddo             ! j
-                 endif
-              endif
+               do j = jlo, jhi
+               do i = ilo, ihi
+                  if (albcnt(i,j,iblk,ns) <= puny) a2D(i,j,n,iblk) = spval_dbl
```
Comment (NickSzapiro-NOAA) [inline cicecore/cicedyn/analysis/ice_history.F90:3676]: Why is there special treatment here? Albedo can still be calculated even if there is no sunlight

Comment (DeniseWorthen) [reply cicecore/cicedyn/analysis/ice_history.F90:3676]: I'm not sure. The average albedo in the history output has always been the average only over hours when the sun is up as far as I know.

**Comment by NickSzapiro-NOAA** [comment]
It's hard for me to understand how cicecore/cicedyn/dynamics/ice_transport_remap.F90 works on the C and CD grids. Is it possible to request documentation like https://cice-consortium-cice.readthedocs.io/en/main/science_guide/sg_horiztrans.html#locating-departure-triangles for these cases?

**Comment by DeniseWorthen** [comment]
I'm not sure what their plan is for further documentation. The general place to ask questions like this is on the CICE Forum https://bb.cgd.ucar.edu/cesm/forums/cice-consortium.146/

File: cicecore/cicedyn/general/ice_init.F90 [hunk 2]
```diff
@@ -443,7 +446,9 @@ subroutine input_data
       ktransport = 1           ! -1 = off, 1 = on
       calc_Tsfc = .true.       ! calculate surface temperature
       update_ocn_f = .false.   ! include fresh water and salt fluxes for frazil
+      cpl_frazil = 'fresh_ice_correction' ! type of coupling for frazil ice
       ustar_min = 0.005        ! minimum friction velocity for ocean heat flux (m/s)
+      hi_min = p01             ! minimum ice thickness allowed (m)
```
Comment (NickSzapiro-NOAA) [inline cicecore/cicedyn/general/ice_init.F90:451]: Is there a rationale for a minimum of 1cm? Consider including a reasonable range in the comment

Comment (DeniseWorthen) [reply cicecore/cicedyn/general/ice_init.F90:451]: I'm not sure I understand your comment. This is "as-is" from the Consortium. New code updates or features can be pushed back to the Consortium, but we don't modify what is in an update in our fork.

File: configuration/scripts/ice_in [hunk 3]
```diff
@@ -125,12 +126,14 @@
...
     conduct           = 'bubbly'
     ksno              = 0.3d0
+    hi_min            = 0.01d0
     a_rapid_mode      =  0.5e-3
     Rac_rapid_mode    =    10.0
     aspect_rapid_mode =     1.0
     dSdt_slow_mode    = -5.0e-8
     phi_c_slow_mode   =    0.05
     phi_i_mushy       =    0.85
+    Tliquidus_max     = -0.1d0
```
Comment (NickSzapiro-NOAA) [inline configuration/scripts/ice_in:136]: Consider Tliquidus_max = 0 as in cicecore/cicedyn/general/ice_init.F90 and doc/source/user_guide/ug_case_settings.rst so they all match

Comment (DeniseWorthen) [reply configuration/scripts/ice_in:136]: This is in the standalone test scripts, unrelated to our use as a coupled component.

**Review by NickSzapiro-NOAA — APPROVED** [review]

**Comment by DeniseWorthen** [comment]
@NeilBarton-NOAA There is a small chance a bug fix in icepack might impact your crash. This will be coming in w/ the latest update on our fork, but you could do a quickie test by changing this one line.

```
diff --git a/columnphysics/icepack_therm_vertical.F90 b/columnphysics/icepack_therm_vertical.F90
index 2d0c20c..bca098e 100644
--- a/columnphysics/icepack_therm_vertical.F90
+++ b/columnphysics/icepack_therm_vertical.F90
@@ -638,7 +638,7 @@ subroutine frzmlt_bottom_lateral (dt,       ncat,     &
       ! FYI: fside is not yet correct for fsd, may need to adjust fbot further
       !-----------------------------------------------------------------

-         xtmp = frzmlt/(fbot + fside + puny)
+         xtmp = frzmlt/(fbot + fside - puny)
          xtmp = min(xtmp, c1)
          fbot  = fbot  * xtmp
          rside = rside * xtmp
```

**Comment by NeilBarton-NOAA** [comment]
@DeniseWorthen Thanks for the awareness.

**Comment by DeniseWorthen** [comment]
@NeilBarton-NOAA Curious if you were able to test this? The PR should go into UWM early next week.

**Comment by NeilBarton-NOAA** [comment]
@DeniseWorthen I did not test it. The crash we are getting is in a part of the code that checks the thickness differences between categories

**Comment by DeniseWorthen** [comment]
The remapping to new categories happens after the vertical thermo, which is why I thought you could be running into this. But that is fine. It is going in because it is an obvious bug.

**Comment by NeilBarton-NOAA** [comment]
I'll see. I'll run a quick test.

**Review by BrianCurtis-NOAA — APPROVED** [review]

**Comment by BrianCurtis-NOAA** [comment]
These CICE PR's are always very large. Do they make a lot of updates at once or do we wait a while before bringing in changes?

**Comment by DeniseWorthen** [comment]
As with MOM6, we try to update ~6weeks. There is no reasonable way to update w/ every Consortium PR, given our pace of commits in UFS.

**Review by junwang-noaa — APPROVED** [review]

**Comment by NeilBarton-NOAA** [comment]
@DeniseWorthen Note, we still get CICE errors with this bug fix. I have another fix for the ICs that I'll present later

**Comment by DeniseWorthen** [comment]
Thanks for checking.

**Pull request merged**